tests/services/upload.service.spec.ts

```diff
@@ -0,0 +1,31 @@
+import {UploadService, SUPPORTED_FILE_TYPES, FILE_TYPE_SUPPORT_ERROR} from '../../src/app/core/services/upload.service';
+
+describe('Test the UploadService', () => {
+    it('Test whether the UploadService can detect valid / invalid filetypes', () => {
+        const uploadService = new UploadService(null);
+        const numOfAcceptedFileTypes = SUPPORTED_FILE_TYPES.length;
+        const randomValidFileType = SUPPORTED_FILE_TYPES[randomIntBetween(0, numOfAcceptedFileTypes)];
+        const validFileName = "testFile." + randomValidFileType;
+        const invalidFileName = "testFile." + 'java';
+        expect(uploadService.isSupportedFileType(validFileName)).toBe(true);
```

Comment:
I think it would be better to iterate through all valid file types, instead of randomly picking one.

Reply:
Yes, that would be better. Done